.clang-format

```diff
@@ -7,35 +8,196 @@
 # behave similar to closing curly braces in a multi-line setting in that
 # they have to be on a line of their own at the same indentation level
 # as the opening part (aka "dangling parenthesis", see https://reviews.llvm.org/D33029).
+#
+# This .clang-format file requires(!) clang-format version 14 or higher
+# in order to support `AlignAfterOpenBracket: BlockIndent`.
 
 Language: Cpp
-BasedOnStyle: LLVM
+Standard: Latest
+BasedOnStyle: Google
 AccessModifierOffset: -4
-AlignAfterOpenBracket: AlwaysBreak
-AlignEscapedNewlinesLeft: true
+AlignAfterOpenBracket: BlockIndent
+AlignArrayOfStructures: None
+AlignConsecutiveMacros: None
+AlignConsecutiveAssignments: None
+AlignConsecutiveBitFields: None
+AlignConsecutiveDeclarations: None
+AlignEscapedNewlines: Left
+# The old PR actually stated false (DontAlign), but seems like Align is correct.
+# i.e. having operators at EOL and the RHS on the next line.
+AlignOperands: Align
```

Comment:
IIUC, ``DontAlign`` was correct. The option is about how far to indent after breaking, isn't it? And we don't want to have operands aligned, but the indent after the operator to be just the usual ``ContinuationIndentWidth``.

But yeah, I just had a quick look at this locally, and there's still a lot off, I'm not sure we can completely fix it.
For example the docs of ``AlignAfterOpenBracket`` in https://clang.llvm.org/docs/ClangFormatStyleOptions.html still state ``Warning Note: This currently only applies to parentheses.`` and we also need it brackets, only to name one...

So given that I'm afraid we'll probably have to postpone this once more.

Reply:
I agree, we don't have to apply it to the sources and *require* the check. However, maybe just having the .clang-format updated as we've come closer.

Reply:
I don't really mind updating the file, but then we should make it a strict improvement, and this line here for example was "more correct" before IIUC. But I'd also say, it's not really worth spending too much time on it, so as far as I'm concerned, I'd just leave things as they are and revisit some other time.

Reply:
I think it actually wouldn't be too bad to merge just the config without the CI part. Currently some contributors or new people do try to use it so having it closer to our style is still of some benefit, even if it does not match perfectly.

Reply:
If we want to merge this, we at least have to change this setting back, though, because that was closer before.

Reply:
They've updated `AlignAfterOpenBracket` to `This applies to round brackets (parentheses), angle brackets and square brackets.`.